src/main/java/org/gridsuite/modification/server/modifications/GenerationDispatch.java

```diff
@@ -0,0 +1,222 @@
+            // get total value of connected loads in the connected component
+            totalDemand.put(componentNum, computeTotalDemand(component, generationDispatchInfos.getLossCoefficient()));
+            report(powerToDispatchReporter, "TotalDemand", "The total demand is : ${totalDemand} MW",
+                Map.of("totalDemand", totalDemand.get(componentNum)), TypedValue.INFO_SEVERITY);
+
+            // get total supply value for non adjustable generators (will be 0. in this first version)
+            double totalAmountFixedSupply = computeTotalAmountFixedSupply(componentNum);
+            report(powerToDispatchReporter, "TotalAmountFixedSupply", "The total amount of fixed supply is : ${totalAmountFixedSupply} MW",
+                Map.of("totalAmountFixedSupply", totalAmountFixedSupply), TypedValue.INFO_SEVERITY);
+
+            double totalAmountSupplyToBeDispatched = totalDemand.get(componentNum) - totalAmountFixedSupply;
+            if (totalAmountSupplyToBeDispatched < 0.) {
+                report(powerToDispatchReporter, "TotalAmountFixedSupplyExceedsTotalDemand", "The total amount of fixed supply exceeds the total demand",
+                    Map.of(), TypedValue.WARN_SEVERITY);
+                continue;
+            } else {
+                report(powerToDispatchReporter, "TotalAmountSupplyToBeDispatched", "The total amount of supply to be dispatched is : ${totalAmountSupplyToBeDispatched} MW",
+                    Map.of("totalAmountSupplyToBeDispatched", totalAmountSupplyToBeDispatched), TypedValue.INFO_SEVERITY);
+            }
+
+            // get adjustable generators in the component
+            computeAdjustableGenerators(component, powerToDispatchReporter);
+            if (adjustableGenerators.get(componentNum).isEmpty()) {
+                continue;
+            }
+
+            // set targetP to 0 for all adjustable generators
+            adjustableGenerators.get(componentNum).forEach(generator -> generator.setTargetP(0.));
+
+            // stacking of adjustable generators to ensure the totalAmountSupplyToBeDispatched
+            List<Scalable> generatorsScalable = adjustableGenerators.get(componentNum).stream().map(generator ->
+                (Scalable) Scalable.onGenerator(generator.getId(), generator.getMinP(), generator.getMaxP())
+            ).collect(Collectors.toList());
+
+            Reporter stackingReporter = componentReporter.createSubReporter(STACKING, STACKING);
+
+            GeneratorTargetPListener listener = new GeneratorTargetPListener(stackingReporter);
+            network.addListener(listener);
+
+            Scalable scalable = Scalable.stack(generatorsScalable.toArray(Scalable[]::new));
+            double realized = scalable.scale(network, totalAmountSupplyToBeDispatched);
+
+            network.removeListener(listener);
+
+            Reporter resultReporter = componentReporter.createSubReporter(RESULT, RESULT);
+
+            if (realized == totalAmountSupplyToBeDispatched) {
```

Comment:
Is strict equality usually achievable ? If unsure, we could be have to resort to `Math.abs(totalAmountSupplyToBeDispatched - realized) < SOME_REASONABLE_EPSILON_I_DONT_KNOW_OF`

Reply:
Done